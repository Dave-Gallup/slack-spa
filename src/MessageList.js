
import { List, Item } from 'semantic-ui-react';
import React, { Component } from 'react';

export default class MessageList extends Component {
  render() {
    return (
      <List aligned="middle" celled="divided">
        <Item
        image="http://semantic-ui.com/images/avatar2/small/lena.png" imageType="image" >
            Lena
        </Item>
        <Item image="http://semantic-ui.com/images/avatar2/small/lena.png" imageType="image" >
            Ann
        </Item>
        <Item image="http://semantic-ui.com/images/avatar2/small/lena.png" imageType="image" >
            Mike
        </Item>
      </List>
    );
  }
}
