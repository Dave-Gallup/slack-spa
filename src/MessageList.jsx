// @flow

import React from 'react';
import { Image, List } from 'semantic-ui-react';
import Message from './Message';
import messages from './messages.json';

export default function MessageList() {
  const messageIds = Object.keys(messages);
  return (
    <List celled >
      {messageIds.map(msgId =>
        <Message
          avatarUrl={messages[msgId].avatarUrl}
          name={messages[msgId].name}
          text={messages[msgId].text}
          timestamp={messages[msgId].timestamp}
        />
      )}
    </List>
  );
}
