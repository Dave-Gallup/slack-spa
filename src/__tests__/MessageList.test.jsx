import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import MessageList from '../MessageList';

describe('Renders message list', () => {
  it('should render an empty message list', () => {
    const messageList = shallow(<MessageList />);
    expect(toJson(messageList)).toMatchSnapshot();
  });
});
