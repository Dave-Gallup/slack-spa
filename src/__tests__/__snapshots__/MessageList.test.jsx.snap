// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders message list should render an empty message list 1`] = `
<List
  celled={true}
>
  <ListItem
    className="listItem"
  >
    <Image
      as="img"
      avatar={true}
      src="http://semantic-ui.com/images/avatar2/small/lena.png"
      ui={true}
    />
    <ListContent>
      <ListHeader>
         Lena Dunham
      </ListHeader>
      <ListDescription>
        I refresh Twitter as thoughtlessly as some twirl their hair.
      </ListDescription>
    </ListContent>
  </ListItem>
  <ListItem
    className="listItem"
  >
    <Image
      as="img"
      avatar={true}
      src="http://semantic-ui.com/images/avatar2/small/matthew.png"
      ui={true}
    />
    <ListContent>
      <ListHeader>
        Matthew McConaughey
      </ListHeader>
      <ListDescription>
        There ain't nothin' more dangerous in this world
                    than a fool with a cause...
      </ListDescription>
    </ListContent>
  </ListItem>
  <ListItem
    className="listItem"
  >
    <Image
      as="img"
      avatar={true}
      src="http://semantic-ui.com/images/avatar2/small/rachel.png"
      ui={true}
    />
    <ListContent>
      <ListHeader>
         Rachel Brosnahan
      </ListHeader>
      <ListDescription>
        I was under the false impression that I could sing in high school, so I did a lot of musical stuff.
      </ListDescription>
    </ListContent>
  </ListItem>
  <ListItem
    className="listItem"
  >
    <Image
      as="img"
      avatar={true}
      src="http://semantic-ui.com/images/avatar/large/jenny.jpg"
      ui={true}
    />
    <ListContent>
      <ListHeader>
        Jennifer Lawrence
      </ListHeader>
      <ListDescription>
        I volunteer! - I gasp. I volunteer as tribute!
      </ListDescription>
    </ListContent>
  </ListItem>
  <ListItem
    className="listItem"
  >
    <Image
      as="img"
      avatar={true}
      src="http://semantic-ui.com/images/avatar/large/stevie.jpg"
      ui={true}
    />
    <ListContent>
      <ListHeader>
        Stevie Nicks
      </ListHeader>
      <ListDescription>
        I am not a good terrified person.
      </ListDescription>
    </ListContent>
  </ListItem>
  <ListItem
    className="listItem"
  >
    <Image
      as="img"
      avatar={true}
      src="http://semantic-ui.com/images/avatar/large/justen.jpg"
      ui={true}
    />
    <ListContent>
      <ListHeader>
        Justin Timberlake
      </ListHeader>
      <ListDescription>
        I got sunshine in my pocket.
      </ListDescription>
    </ListContent>
  </ListItem>
</List>
`;
